### PR TITLE
demo pixi.js

### PR DESCRIPTION
**Pull request by kostmo**
Towards #1415.

Potential candidate for a web-based world display

Uses [pixi.js](https://pixijs.com/).

# Demo

1. Run swarm: `scripts/play.sh -i scenarios/Fun/horton.yaml --autoplay --speed 6`
1. Visit http://localhost:5357/play.html

### CODE REVIEW

File: src/swarm-util/Swarm/Util/OccurrenceEncoder.hs [hunk 1]
```diff
@@ -0,0 +1,47 @@
+-- |
+-- SPDX-License-Identifier: BSD-3-Clause
+-- Description: Compress representation of traversable
+--
+-- For structures with many repeating elements,
+-- will store the expanded element representation
+-- in an array so that the structure can be transmitted
+-- with array indices as elements.
```
Comment (byorgey) [inline src/swarm-util/Swarm/Util/OccurrenceEncoder.hs:8]: This description and comment did not make much sense to me.  What is a "traversable" in this context?  What is a "structure"?  What is an "expanded element representation"?  It seems to me that this module is doing something fairly straightforward: keep track of seen items with a unique index for each, allowing us to quickly look up the index of an item or assign a new index to an item we have not seen before.  I'm guessing that you were writing these descriptions with the *intended use case* in mind, but I think prior to that we should just write a simple description of what it actually does (and then optionally include a description of how it is used).

File: web/play.html [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        doFetch(button, appView, graphics, displayWidth, displayHeight);
+      }
+
+      window.onload=()=>{
+        startLoop();
+      }
+    </script>
+  </head>
+  <body>
+    <button id="restart-button" onclick="startLoop();" style="display: block;">Restart</button>
```
Comment (byorgey) [inline web/play.html:30]: When I run the web frontend I just see an image displaying a view of the world; I don't see any kind of "Restart" button.  I'm not sure I understand whether that is the expected behavior.

**Review by byorgey — APPROVED** [review]

File: web/play.html (hunk 2 above)
Comment (kostmo) [reply web/play.html:30]: Yeah, the restart button will (or is supposed to, anyway) appear only if an error is encountered.

File: src/swarm-util/Swarm/Util/OccurrenceEncoder.hs (hunk 1 above)
Comment (kostmo) [reply src/swarm-util/Swarm/Util/OccurrenceEncoder.hs]: Yes, the struggle with module documentation was a sign that the API was not good.  I've simplified it a bit.

**Pull request merged**